lib/src/nekoton_webview.dart

```diff
@@ -1,7 +1,47 @@
-/// {@template nekoton_webview}
-/// Flutter webview nekoton inpage provider bindings
-/// {@endtemplate}
-class NekotonWebview {
-  /// {@macro nekoton_webview}
-  const NekotonWebview();
+import 'package:flutter/services.dart';
+import 'package:flutter_inappwebview/flutter_inappwebview.dart';
+
+///
+extension NekotonWebview on InAppWebViewController {
+  ///Initialize inpage provider
+  ///```dart
+  /// InAppWebView(
+  ///   onWebViewCreated: (InAppWebViewController controller) async {
+  ///     await controller.initNekotonProvider(
+  ///       handler: ({required String method, dynamic params}) {
+  ///         print({method, params});
+  ///       }
+  ///     );
+  ///   }
+  /// )
+  ///```
+  Future<void> initNekotonProvider({
+    required NekotonProviderHandler handler,
+  }) async {
+    await addUserScript(
+      userScript: UserScript(
+        source: await rootBundle
+            .loadString('packages/nekoton_webview/assets/main.js'),
+        injectionTime: UserScriptInjectionTime.AT_DOCUMENT_START,
+      ),
+    );
+
+    addJavaScriptHandler(
+      handlerName: 'nekotonWebviewHandler',
+      callback: (List<dynamic> arguments) {
```

Comment:
It's better to create a clear interface by providing an abstract class in the library and forcing user to pass an instance of inherited class here. Thus, the library user will have to implement the required methods with the necessary arguments, and also receive documentation for each of them (yeah, docs please?).